libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
I still fail to see why it needs to be this way. Outside modules should do:

```
compass = AP_Compass::get_instance();
```

And then use it like any other users. The advantages are:
 - this is still giving you a static object you can use and the users don't need to care about it
 - data is still part of the object - otherwise it would be better to just have a namespace rather than a class and hide the data inside the .cpp

> @tridge has informed me that InertialSensor should also be converted to have a static object of itself instead of static pointer to object of itself.

This is not what we have in the code. We do have a static object. And we guarantee there's only one instance at any time. The change went in probably without @tridge noticing it - it's already a singleton.